src/components/EditTab/index.js

```diff
@@ -49,160 +32,45 @@ export default class EditTab extends PureComponent {
 
-    const disableSave = (formInstance && isEqual(persistentInstance, formInstance));
-
-    const buttons = [];
-
-    if (exitView) {
-      buttons.push(
-        <ConfirmUnsavedChanges key='Cancel' showDialog={this.hasUnsavedChanges}>
+    const buttons = [
+      ...(!!exitView && [
+        <ConfirmUnsavedChanges key='Cancel' showDialog={_ => unsavedChanges}>
           <Button bsStyle='link' onClick={exitView}>
             {i18n.getMessage('crudEditor.cancel.button')}
           </Button>
         </ConfirmUnsavedChanges>
-      )
-    }
-
-    if (persistentInstance) {
-      buttons.push(
-        ...internalOperations(persistentInstance).map(({ name, icon, handler, type }, index) => (
-          <ConfirmUnsavedChanges
-            key={`internal-operation-${index}`}
-            showDialog={this.showConfirmDialog}
-          >
-            <Button onClick={handler}>
-              {icon && <Glyphicon glyph={icon} />}
-              {icon && ' '}
-              {getModelMessage({ i18n, key: `model.label.${name}`, defaultMessage: titleCase(name) })}
-            </Button>
-          </ConfirmUnsavedChanges>
-        ))
-      );
-    }
-
-    buttons.push(
-      externalOperations.map(({ title, icon, handler }, index) => (
-        <Button
-          onClick={_ => handler(persistentInstance)}
-          key={`external-operation-${index}`}
-        >
-          {icon && <Glyphicon glyph={icon} />}
-          {icon && ' '}
-          {title}
-        </Button>
-      ))
-    )
-
-    if (deleteInstances) {
-      buttons.push(
-        <ConfirmDialog
-          message={i18n.getMessage('crudEditor.delete.confirmation')}
-          textConfirm={i18n.getMessage('crudEditor.delete.button')}
-          textCancel={i18n.getMessage('crudEditor.cancel.button')}
-          key="Delete"
-        >
-          <Button
-            onClick={this.handleDelete}
-            disabled={!!deleteConfig.disabled}
-          >
-            {i18n.getMessage('crudEditor.delete.button')}
-          </Button>
-        </ConfirmDialog>
-      )
-    }
-
-    if (saveAndNewInstance) {
-      buttons.push(
-        <Button
-          onClick={this.handleSaveAndNew}
-          disabled={disableSave}
-          key="Save and New"
-        >
-          {i18n.getMessage('crudEditor.saveAndNew.button')}
-        </Button>)
-    }
-
-    if (saveAndNextInstance) {
-      buttons.push(
-        <Button
-          onClick={this.handleSaveAndNext}
-          disabled={disableSave}
-          key="Save and Next"
-        >
-          {i18n.getMessage('crudEditor.saveAndNext.button')}
-        </Button>)
-    }
-
-    if (saveInstance) {
-      buttons.push(
-        <Button
-          disabled={disableSave}
-          bsStyle='primary'
-          type='submit'
-          key="Save"
-        >
+      ]),
+      ...getOperationButtons({ operations }),
+      ...(!!saveInstance && [
+        <Button disabled={!unsavedChanges} bsStyle='primary' type='submit' key='Save'>
```

Comment:
Why not move 'cancel' and 'save' buttons to operations list defined in 'container' file?